testing/src/main/java/org/oppia/android/testing/media/TestMediaPlayerRobolectricImpl.kt

```diff
@@ -0,0 +1,169 @@
+
+  override fun addMediaInfo(
+    context: Context,
+    explorationId1: String,
+    explorationId2: String,
+    audioFileName1: String,
+    audioFileName2: String,
+    mediaPlayer: MediaPlayer
+  ) {
+    val mediaTestUrl1 = createAudioUrl(explorationId1, audioFileName1)
+    val mediaTestUrl2 = createAudioUrl(explorationId2, audioFileName2)
+
+    val dataSource = toDataSource(context, Uri.parse(mediaTestUrl1))!!
+    val dataSource2 = toDataSource(context, Uri.parse(mediaTestUrl2))!!
+    val mediaInfo = createMediaInfo(
+      duration = 1000,
+      preparationDelay = 0
+    )
+    addMediaInfo(dataSource, mediaInfo)
+    addMediaInfo(dataSource2, mediaInfo)
+
+    val shadowMediaPlayer = shadowOf(mediaPlayer)!!
+    setDataSource(shadowMediaPlayer, toDataSource(context, Uri.parse(mediaTestUrl1))!!)
+
+    invokePreparedListener(shadowMediaPlayer)
+  }
+
+  /** Calls DataSource.toDataSource() using reflection. */
+  private fun toDataSource(context: Context, uri: Uri): Any? {
+    val dataSourceClass = getDataSourceClass()
+
+    val toDataSourceMethod =
+      dataSourceClass.getMethod("toDataSource", Context::class.java, Uri::class.java)
+    return toDataSourceMethod.invoke(/* obj= */ null, context, uri)
+  }
+
+  /** Returns a new ShadowMediaPlayer.MediaInfo using reflection. */
+  private fun createMediaInfo(duration: Int, preparationDelay: Int): Any {
+    val mediaInfoClass = getMediaInfoClass()
+
+    return mediaInfoClass
+      .getConstructor(Int::class.java, Int::class.java)
+      .newInstance(duration, preparationDelay)
+  }
+
+  /** Calls ShadowMediaPlayer.addMediaInfo() using reflection. */
+  private fun addMediaInfo(dataSource: Any, mediaInfo: Any) {
+    val shadowMediaPlayerClass = getShadowMediaPlayerClass()
+
+    val dataSourceClass = getDataSourceClass()
+
+    val mediaInfoClass = getMediaInfoClass()
+
+    val addMediaInfoMethod =
+      shadowMediaPlayerClass.getMethod("addMediaInfo", dataSourceClass, mediaInfoClass)
+
+    addMediaInfoMethod.invoke(/* obj= */ null, dataSource, mediaInfo)
+  }
+
+  /** Calls Robolectric's Shadows.shadowOf() using reflection. */
+  private fun shadowOf(mediaPlayer: MediaPlayer): Any? {
+    val shadowsClass = getShadowsClass()
+
+    return shadowsClass.getMethod("shadowOf", MediaPlayer::class.java)
+      .invoke(/* obj= */ null, mediaPlayer)
+  }
+
+  /** Calls ShadowMediaPlayer.setDataSource() using reflection. */
+  private fun setDataSource(shadowMediaPlayer: Any, dataSource: Any) {
+    val dataSourceClass = getDataSourceClass()
+
+    shadowMediaPlayer.javaClass.getMethod("setDataSource", dataSourceClass)
+      .invoke(shadowMediaPlayer, dataSource)
+  }
+
+  /** Calls ShadowMediaPlayer.invokePreparedListener() using reflection. */
+  private fun invokePreparedListener(shadowMediaPlayer: Any) {
+    shadowMediaPlayer.javaClass.getMethod("invokePreparedListener").invoke(shadowMediaPlayer)
+  }
+
+  override fun setupAudio(explorationId: String, audioFileName: String) {
```

Comment:
``setUp`` rather than ``setup`` here & elsewhere, as well.

Reply:
Done

Reply:
resolving on behalf of Ben